src/WalletLink.ts

```diff
@@ -185,10 +173,21 @@ export class WalletLink {
    * all potential stale state is cleared.
    */
   public disconnect(): void {
-    if (typeof window.walletLinkExtension !== "undefined") {
-      window.walletLinkExtension.close()
+    const extension = this.getCBWalletExtension()
+    if (extension) {
+      extension.close()
     } else {
       this._relay?.resetAndReload()
     }
   }
+
+  private getCBWalletExtension(): WalletLinkProvider | undefined {
+    return window.walletLinkExtension
+  }
+
+  private isCipherProvider(provider: WalletLinkProvider): boolean {
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    return typeof provider.isCipher === "boolean" && provider.isCipher
```

Comment:
Is `isCipher` used anymore? I don't see any reference to it in cb wallet.

Reply:
correct. it's not in our current code base. i think this is for backward compatibility to support old clients who haven't updated yet.